packages/frontend/src/pages/admin/bot-protection.vue

```diff
@@ -138,22 +198,52 @@ const botProtectionForm = useForm({
 	turnstileSiteKey: meta.turnstileSiteKey,
 	turnstileSecretKey: meta.turnstileSecretKey,
 }, async (state) => {
-	await os.apiWithDialog('admin/update-meta', {
-		enableHcaptcha: state.provider === 'hcaptcha',
-		hcaptchaSiteKey: state.hcaptchaSiteKey,
-		hcaptchaSecretKey: state.hcaptchaSecretKey,
-		enableMcaptcha: state.provider === 'mcaptcha',
-		mcaptchaSiteKey: state.mcaptchaSiteKey,
-		mcaptchaSecretKey: state.mcaptchaSecretKey,
-		mcaptchaInstanceUrl: state.mcaptchaInstanceUrl,
-		enableRecaptcha: state.provider === 'recaptcha',
-		recaptchaSiteKey: state.recaptchaSiteKey,
-		recaptchaSecretKey: state.recaptchaSecretKey,
-		enableTurnstile: state.provider === 'turnstile',
-		turnstileSiteKey: state.turnstileSiteKey,
-		turnstileSecretKey: state.turnstileSecretKey,
-		enableTestcaptcha: state.provider === 'testcaptcha',
-	});
-	fetchInstance(true);
+	const provider = botProtectionForm.state.provider;
+
+	const sitekey = provider === 'hcaptcha'
+		? botProtectionForm.state.hcaptchaSiteKey
+		: provider === 'mcaptcha'
+			? botProtectionForm.state.mcaptchaSiteKey
+			: provider === 'recaptcha'
+				? botProtectionForm.state.recaptchaSiteKey
+				: provider === 'turnstile'
+					? botProtectionForm.state.turnstileSiteKey
+					: null;
+	const secret = provider === 'hcaptcha'
+		? botProtectionForm.state.hcaptchaSecretKey
+		: provider === 'mcaptcha'
+			? botProtectionForm.state.mcaptchaSecretKey
+			: provider === 'recaptcha'
+				? botProtectionForm.state.recaptchaSecretKey
+				: provider === 'turnstile'
+					? botProtectionForm.state.turnstileSecretKey
+					: null;
+
+	if (captchaResult.value) {
+		await os.apiWithDialog(
+			'admin/captcha/save',
+			{
+				provider: provider as Misskey.entities.AdminCaptchaSaveRequest['provider'],
+				sitekey: sitekey,
+				secret: secret,
+				instanceUrl: botProtectionForm.state.mcaptchaInstanceUrl,
+				captchaResult: captchaResult.value,
+			},
+		);
```

Comment:
```suggestion
		await os.apiWithDialog('admin/captcha/save', {
			provider: provider as Misskey.entities.AdminCaptchaSaveRequest['provider'],
			sitekey: sitekey,
			secret: secret,
			instanceUrl: botProtectionForm.state.mcaptchaInstanceUrl,
			captchaResult: captchaResult.value,
		}, undefined, {
			'14bf7ae1-80cc-4363-acb2-4fd61d086af0': {
				title: 'このCAPTCHAシステムには対応していません',
				text: '設定をお確かめください',
			},
			// ...
		});
```

的なことをやるとエラーハンドリングができる